Remove most EnumTraits specializations under WebKit/Shared

zdobersek opened the pull request:
#### d82d99d57d3f36d727cbc1e6c60155a42d81818a
<pre>
Remove most EnumTraits specializations under WebKit/Shared
<a href="https://bugs.webkit.org/show_bug.cgi?id=266587">https://bugs.webkit.org/show_bug.cgi?id=266587</a>

Reviewed by Chris Dumez.

Remove the majority of EnumTraits specializations under WebKit/Shared in favor
of serialization specifications that will be used to automatically generate the
relevant enum validation code.

* Source/WebKit/CMakeLists.txt:
* Source/WebKit/DerivedSources-input.xcfilelist:
* Source/WebKit/DerivedSources.make:
* Source/WebKit/Shared/Authentication/AuthenticationChallengeDisposition.h:
* Source/WebKit/Shared/Authentication/AuthenticationChallengeDisposition.serialization.in: Copied from Source/WebKit/Shared/LayerTreeContext.serialization.in.
* Source/WebKit/Shared/BackgroundFetchChange.h:
* Source/WebKit/Shared/BackgroundFetchChange.serialization.in: Copied from Source/WebKit/Shared/LayerTreeContext.serialization.in.
* Source/WebKit/Shared/CacheModel.h:
* Source/WebKit/Shared/CacheModel.serialization.in: Copied from Source/WebKit/Shared/LayerTreeContext.serialization.in.
* Source/WebKit/Shared/CoordinateSystem.h:
* Source/WebKit/Shared/CoordinateSystem.serialization.in: Copied from Source/WebKit/Shared/LayerTreeContext.serialization.in.
* Source/WebKit/Shared/DragControllerAction.h:
* Source/WebKit/Shared/DragControllerAction.serialization.in: Copied from Source/WebKit/Shared/LayerTreeContext.serialization.in.
* Source/WebKit/Shared/FocusedElementInformation.h:
* Source/WebKit/Shared/FocusedElementInformation.serialization.in:
* Source/WebKit/Shared/InspectorExtensionTypes.h:
* Source/WebKit/Shared/InspectorExtensionTypes.serialization.in: Copied from Source/WebKit/Shared/LayerTreeContext.serialization.in.
* Source/WebKit/Shared/LayerTreeContext.h:
* Source/WebKit/Shared/LayerTreeContext.serialization.in:
* Source/WebKit/Shared/MediaPlaybackState.h:
* Source/WebKit/Shared/MediaPlaybackState.serialization.in: Copied from Source/WebKit/Shared/LayerTreeContext.serialization.in.
* Source/WebKit/Shared/ResourceLoadInfo.h:
* Source/WebKit/Shared/SyntheticEditingCommandType.h:
* Source/WebKit/Shared/SyntheticEditingCommandType.serialization.in: Copied from Source/WebKit/Shared/LayerTreeContext.serialization.in.
* Source/WebKit/Shared/TextRecognitionUpdateResult.h:
* Source/WebKit/Shared/TextRecognitionUpdateResult.serialization.in: Copied from Source/WebKit/Shared/LayerTreeContext.serialization.in.
* Source/WebKit/Shared/UndoOrRedo.h:
* Source/WebKit/Shared/UndoOrRedo.serialization.in: Copied from Source/WebKit/Shared/LayerTreeContext.serialization.in.
* Source/WebKit/Shared/VideoCodecType.h:
* Source/WebKit/Shared/VideoCodecType.serialization.in: Copied from Source/WebKit/Shared/LayerTreeContext.serialization.in.
* Source/WebKit/Shared/WebGPU/WebGPUBindGroupEntry.h:
* Source/WebKit/Shared/WebGPU/WebGPUBindGroupEntry.serialization.in:
* Source/WebKit/Shared/WebsiteAutoplayPolicy.h:
* Source/WebKit/Shared/WebsiteAutoplayPolicy.serialization.in: Copied from Source/WebKit/Shared/LayerTreeContext.serialization.in.
* Source/WebKit/Shared/WebsiteAutoplayQuirk.h:
* Source/WebKit/Shared/WebsiteAutoplayQuirk.serialization.in: Copied from Source/WebKit/Shared/LayerTreeContext.serialization.in.
* Source/WebKit/Shared/WebsiteData/UnifiedOriginStorageLevel.h:
* Source/WebKit/Shared/WebsiteData/UnifiedOriginStorageLevel.serialization.in: Copied from Source/WebKit/Shared/LayerTreeContext.serialization.in.
* Source/WebKit/Shared/WebsitePopUpPolicy.h:
* Source/WebKit/Shared/WebsitePopUpPolicy.serialization.in: Copied from Source/WebKit/Shared/LayerTreeContext.serialization.in.
* Source/WebKit/WebKit.xcodeproj/project.pbxproj:

Canonical link: <a href="https://commits.webkit.org/272269@main">https://commits.webkit.org/272269@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/97d5af859f063f282c5bdaabfe995d3843ec8633

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/31132 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/9804 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/32820 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/33637 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/28096 "Built successfully") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/12161 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/51/builds/7062 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/27933 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/31464 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/8276 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/27827 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/7098 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/7283 "Passed tests") | | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/34977 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/28344 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/28179 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/33402 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/7306 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/50/builds/5362 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/31240 "Passed tests") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/8993 "Built successfully") | | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/7329 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/8010 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/7840 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/015eee8d40e6b4594bd9ac4c8aac1e7745d733e3. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/21976

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/97d5af859f063f282c5bdaabfe995d3843ec8633. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/21976

zdobersek committed:
Remove most EnumTraits specializations under WebKit/Shared
https://bugs.webkit.org/show_bug.cgi?id=266587

Reviewed by Chris Dumez.

Remove the majority of EnumTraits specializations under WebKit/Shared in favor
of serialization specifications that will be used to automatically generate the
relevant enum validation code.

* Source/WebKit/CMakeLists.txt:
* Source/WebKit/DerivedSources-input.xcfilelist:
* Source/WebKit/DerivedSources.make:
* Source/WebKit/Shared/Authentication/AuthenticationChallengeDisposition.h:
* Source/WebKit/Shared/Authentication/AuthenticationChallengeDisposition.serialization.in: Copied from Source/WebKit/Shared/LayerTreeContext.serialization.in.
* Source/WebKit/Shared/BackgroundFetchChange.h:
* Source/WebKit/Shared/BackgroundFetchChange.serialization.in: Copied from Source/WebKit/Shared/LayerTreeContext.serialization.in.
* Source/WebKit/Shared/CacheModel.h:
* Source/WebKit/Shared/CacheModel.serialization.in: Copied from Source/WebKit/Shared/LayerTreeContext.serialization.in.
* Source/WebKit/Shared/CoordinateSystem.h:
* Source/WebKit/Shared/CoordinateSystem.serialization.in: Copied from Source/WebKit/Shared/LayerTreeContext.serialization.in.
* Source/WebKit/Shared/DragControllerAction.h:
* Source/WebKit/Shared/DragControllerAction.serialization.in: Copied from Source/WebKit/Shared/LayerTreeContext.serialization.in.
* Source/WebKit/Shared/FocusedElementInformation.h:
* Source/WebKit/Shared/FocusedElementInformation.serialization.in:
* Source/WebKit/Shared/InspectorExtensionTypes.h:
* Source/WebKit/Shared/InspectorExtensionTypes.serialization.in: Copied from Source/WebKit/Shared/LayerTreeContext.serialization.in.
* Source/WebKit/Shared/LayerTreeContext.h:
* Source/WebKit/Shared/LayerTreeContext.serialization.in:
* Source/WebKit/Shared/MediaPlaybackState.h:
* Source/WebKit/Shared/MediaPlaybackState.serialization.in: Copied from Source/WebKit/Shared/LayerTreeContext.serialization.in.
* Source/WebKit/Shared/ResourceLoadInfo.h:
* Source/WebKit/Shared/SyntheticEditingCommandType.h:
* Source/WebKit/Shared/SyntheticEditingCommandType.serialization.in: Copied from Source/WebKit/Shared/LayerTreeContext.serialization.in.
* Source/WebKit/Shared/TextRecognitionUpdateResult.h:
* Source/WebKit/Shared/TextRecognitionUpdateResult.serialization.in: Copied from Source/WebKit/Shared/LayerTreeContext.serialization.in.
* Source/WebKit/Shared/UndoOrRedo.h:
* Source/WebKit/Shared/UndoOrRedo.serialization.in: Copied from Source/WebKit/Shared/LayerTreeContext.serialization.in.
* Source/WebKit/Shared/VideoCodecType.h:
* Source/WebKit/Shared/VideoCodecType.serialization.in: Copied from Source/WebKit/Shared/LayerTreeContext.serialization.in.
* Source/WebKit/Shared/WebGPU/WebGPUBindGroupEntry.h:
* Source/WebKit/Shared/WebGPU/WebGPUBindGroupEntry.serialization.in:
* Source/WebKit/Shared/WebsiteAutoplayPolicy.h:
* Source/WebKit/Shared/WebsiteAutoplayPolicy.serialization.in: Copied from Source/WebKit/Shared/LayerTreeContext.serialization.in.
* Source/WebKit/Shared/WebsiteAutoplayQuirk.h:
* Source/WebKit/Shared/WebsiteAutoplayQuirk.serialization.in: Copied from Source/WebKit/Shared/LayerTreeContext.serialization.in.
* Source/WebKit/Shared/WebsiteData/UnifiedOriginStorageLevel.h:
* Source/WebKit/Shared/WebsiteData/UnifiedOriginStorageLevel.serialization.in: Copied from Source/WebKit/Shared/LayerTreeContext.serialization.in.
* Source/WebKit/Shared/WebsitePopUpPolicy.h:
* Source/WebKit/Shared/WebsitePopUpPolicy.serialization.in: Copied from Source/WebKit/Shared/LayerTreeContext.serialization.in.
* Source/WebKit/WebKit.xcodeproj/project.pbxproj:

Canonical link: https://commits.webkit.org/272269@main

webkit-commit-queue commented:
Committed 272269@main (d82d99d57d3f): <https://commits.webkit.org/272269@main>

Reviewed commits have been landed. Closing PR #21976 and removing active labels.
